### PR TITLE
Support Australian customers to use Australian Stripe

### PR DESCRIPTION
**Pull request by lmath**
Some Australian subscribers have been incurring a transaction fee, so here we switch Australian customers to use our Australian stripe account.

Changes:

- Add an AUStripeService to the PaymentService 
- Pick which stripe service to call based on billing country
- Add the public keys to the javascriptFirstSteps template
- Add which stripe service to communicate with to the CountryWithCurrency and then use this information to determine which public key to use when the Stripe checkout is launched.

cc @AWare @paulbrown1982 @pvighi @jacobwinch @pvighi

### CODE REVIEW

File: app/controllers/Checkout.scala [hunk 1]
```diff
@@ -33,10 +33,12 @@ import views.support.{PlanList, BillingPeriod => _, _}
 
 import scala.Function.const
 import play.api.libs.concurrent.Execution.Implicits.defaultContext
+
 import scala.concurrent.Future
 import scalaz.std.option._
 import scalaz.std.scalaFuture._
 import scalaz.syntax.applicative._
+import scalaz.{NonEmptyList, OptionT, \/}
 import scalaz.{NonEmptyList, OptionT}
```
Comment (jacobwinch) [inline app/controllers/Checkout.scala]: Duplicate imports here?

File: app/controllers/Checkout.scala [hunk 2]
```diff
@@ -174,7 +176,7 @@ object Checkout extends Controller with LazyLogging with CatalogProvider {
     implicit val resolution: TouchpointBackend.Resolution = TouchpointBackend.forRequest(NameEnteredInForm, tempData)
     implicit val tpBackend = resolution.backend
     val idUserOpt = authenticatedUserFor(request)
-    val srEither = tpBackend.subsForm.bindFromRequest
+    val srEither: \/[Seq[FormError], SubscribeRequest] = tpBackend.subsForm.bindFromRequest
```
Comment (jacobwinch) [inline app/controllers/Checkout.scala]: I know this hasn't been changed here, but 'srEither' and 'sr' are not great names...

File: app/services/CheckoutService.scala [hunk 3]
```diff
@@ -345,7 +347,7 @@ class CheckoutService(
       val idMinimalUser = IdMinimalUser(contact.identityId, None)
       val purchaserIds = PurchaserIdentifiers(contact, Some(idMinimalUser))
       renewal.paymentData match {
-        case cd: CreditCardData => paymentService.makeZuoraAccountWithCreditCard(cd, subscription.currency, purchaserIds)
+        case cd: CreditCardData => paymentService.makeZuoraAccountWithCreditCard(cd, subscription.currency, purchaserIds,billto.country)
```
Comment (jacobwinch) [inline app/services/CheckoutService.scala]: Minor niggle, but we're missing a space here

File: app/views/support/CountryWithCurrency.scala [hunk 4]
```diff
@@ -5,12 +5,21 @@ import com.gu.i18n.{Country, CountryGroup, Currency}
...
 
-case class CountryWithCurrency(country: i18n.Country, currency: i18n.Currency)
+sealed trait StripeServiceName {
+  val jsLookupKey: String
+}
+case object UKStripeService extends StripeServiceName {
+  val jsLookupKey = "ukPublicKey" // Referenced in assets/javascripts/modules/checkout/submit.js
+}
+case object AUStripeService extends StripeServiceName {
+  val jsLookupKey = "auPublicKey"
```
Comment (jacobwinch) [inline app/views/support/CountryWithCurrency.scala]: Is this used?

File: app/views/fragments/javascriptFirstSteps.scala.html [hunk 5]
```diff
@@ -37,8 +38,12 @@
         touchpointBackendResolution <- touchpointBackendResolutionOpt;
         backend = touchpointBackendResolution.backend
     ) {
+    guardian.stripe = {
+        '@{UKStripeService.jsLookupKey}': '@backend.stripeUKMembershipService.publicKey.mkString',
+        '@{AUStripeService.jsLookupKey}': '@backend.stripeAUMembershipService.publicKey.mkString'
+    };
     guardian.stripeCheckout = {
-        key: '@backend.paymentService.stripeService.publicKey',
+        key: guardian.stripe['@{UKStripeService.jsLookupKey}'],
```
Comment (jacobwinch) [inline app/views/fragments/javascriptFirstSteps.scala.html]: Is something being hardcoded to use the main (UK) Stripe account here?

Comment (paulbrown1982) [reply app/views/fragments/javascriptFirstSteps.scala.html]: As a fallback; it gets overridden in submit.js

File: app/views/support/CountryWithCurrency.scala (hunk 4 above)
Comment (paulbrown1982) [reply app/views/support/CountryWithCurrency.scala]: Line 43 of javascriptFirstSteps.scala.html

File: app/views/fragments/javascriptFirstSteps.scala.html (hunk 5 above)
Comment (jacobwinch) [reply app/views/fragments/javascriptFirstSteps.scala.html]: Ah I see - thanks

**Comment by jacobwinch** [comment]
Looks good to me - just a couple of minor comments 👍

File: app/views/support/CountryWithCurrency.scala (hunk 4 above)
Comment (paulbrown1982) [reply app/views/support/CountryWithCurrency.scala]: Fixed

**Comment by paulbrown1982** [comment]
Tested in prod.

Before:
picture 242

After:
picture 243

**Pull request merged**